packages/core/src/application.ts

```diff
@@ -133,6 +139,9 @@ export class Application extends Context {
   private _setupOperation(verb: string, path: string, spec: OperationObject) {
     const handler = spec['x-operation'];
     if (typeof handler === 'function') {
+      spec = Object.assign({}, spec);
```

Comment:
It depends on how defensive we want to be. I am creating a shallow copy of the spec object because I want to delete `x-operation` but I don't want to modify the original value provided by the user. For that purpose, shallow copy via `Object.assign` is enough.